ui/pages/confirmations/components/simulation-details/formatAmount.ts

```diff
@@ -80,7 +80,8 @@ export function formatAmount(locale: string, amount: BigNumber): string {
     MAX_SIGNIFICANT_DECIMAL_PLACES - digitsLeftOfDecimal + 1,
   );
 
-  return new Intl.NumberFormat(locale, {
-    maximumFractionDigits,
-  } as Intl.NumberFormatOptions).format(amount.toNumber());
+  return amount.toFormat(maximumFractionDigits);
```

Comment:
Hi @jpuri! To preserve the internationalization but still show all decimals, you could use BigInt.toLocaleString:

```js
if (maximumFractionDigits === 0) {
  // No decimals to display – we can use BigInt to localize without loss of precision.
  return BigInt(amount.toFixed());
} else {
  // At most 4 (MAX_SIGNIFICANT_DECIMAL_PLACES + 1) significant digits – Use Intl.NumberFormat to localize
  return new Intl.NumberFormat(locale, {
      maximumFractionDigits,
    } as Intl.NumberFormatOptions).format(amount.toNumber());
}
```

Here's an example:
```js
  const bigNum = BigNumber(
    "115792089237316195423570985008687907853269984665640564039457584007913129639935"
  );
  const bigInt = BigInt(bigNum.toFixed());
  console.log(bigInt.toLocaleString());
  // 115,792,089,237,316,195,423,570,985,008,687,907,853,269,984,665,640,564,039,457,584,007,913,129,639,935

```

Reply:
Hey @dbrans : we still lose precision for long values with decimals.

Reply:
Hey Jyoti, the loss of decimal places for large numbers is [intentional](https://github.com/MetaMask/metamask-extension/blob/694ecf8d583041d391e4028e65f6092eaa5369c3/ui/pages/confirmations/components/simulation-details/formatAmount.ts#L28-L31). This was decisionwas made  by @bschorchit and Saya [here](https://www.notion.so/Simulation-Release-Candidate-Testing-Feedback-7dd07ced652f4a2b8dc2ae96cedc69a4?d=15202499537e4404bdaf38f67c4803dd&pvs=4#930c7e1aef8049caa203953fa88dd23a).

Reply:
for instance this number:

`1157920892373161954235709850086879078532699846656405640394575.84007913129639935`

formats as `1,157,920,892,373,162,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000`

Reply:
Jyoti, you may have to play with it since we use a different version on BigNumber, but here is a [sandbox where it is working](https://codesandbox.io/p/sandbox/bignumber-js-forked-3nvlft?file=%2Fsrc%2Findex.js).
```js
  const bigNum = BigNumber(
    "1157920892373161954235709850086879078532699846656405640394575.84007913129639935"
  );
  const bigInt = BigInt(bigNum.toFixed(0));
  console.log(bigInt.toLocaleString("fr"));
  // 1 157 920 892 373 161 954 235 709 850 086 879 078 532 699 846 656 405 640 394 576
```

Reply:
I see ur point, I updated the PR. Please review.

Reply:
Hey @dbrans : I found that this fix is working in jest but for reason blowing up in extension.